chia/simulator/block_tools.py

```diff
@@ -540,6 +542,29 @@ def get_farmer_wallet_tool(self) -> WalletTool:
     def get_pool_wallet_tool(self) -> WalletTool:
         return WalletTool(self.constants, self.pool_master_sk)
 
+    # Verifies if the given plot passed any of the previous `UNIQUE_PLOTS_WINDOW` plot filters.
+    def plot_id_passed_previous_filters(self, plot_id: bytes32, cc_sp_hash: bytes32, blocks: List[FullBlock]) -> bool:
+        curr_sp_hash = cc_sp_hash
+        sp_count = 1
+        for block in reversed(blocks):
```

Comment:
is this cheap to do?

Reply:
I _think_ it's a reverse iterator - so it won't do any reversing. also, the loop breaks really fast (when we hit 4 distinct SP hashes), so we won't iterate almost at all through blocks.